<launch>
  <arg name="INPUT_IMAGE"/>
  <arg name="IMAGE_TRANSPORT" default="raw"/>
  <arg name="model_file" default="$(find coral_usb)/python/coral_usb/posenet/models/posenet_mobilenet_v1_075_481_641_quant_decoder_edgetpu.tflite"/>
  <arg name="device_id" default="0" />
  <arg name="machine" default="c2" />
  <machine name="c1" address="c1" default="true"/>
  <machine name="c2" address="c2" env-loader="$(find coral_usb)/launch/env.sh"/>

  <node name="edgetpu_human_pose_estimator"
        pkg="coral_usb" type="edgetpu_human_pose_estimator.py"
        output="screen" respawn="true" machine="$(arg machine)">
    <remap from="~input" to="$(arg INPUT_IMAGE)" />
    <rosparam subst_value="true" >
      model_file: $(arg model_file)
      image_transport: $(arg IMAGE_TRANSPORT)
      device_id: $(arg device_id)
    </rosparam>
  </node>
</launch>
